src/docs/common/snippets/common-graalpy-graalvm.adoc

```diff
@@ -0,0 +1,71 @@
+:exclude-for-languages:groovy
+
+== Python Runtime Compilation
+
+With GraalVM JDK, the Python code executed using the GraalPy engine will be just-in-time compiled.
+With other JDKs, the Python code will be interpreted, which negatively impacts its performance.
+
+The easiest way to install https://www.graalvm.org[GraalVM] on Linux or Mac is to use https://sdkman.io/[SDKMan.io].
+
+[source, bash]
+.Java 23
+----
+sdk install java 23-graal
+----
+
+For installation on Windows, or for a manual installation on Linux or Mac, see the https://www.graalvm.org/latest/docs/getting-started/[GraalVM Getting Started] documentation.
+
+The previous command installs Oracle GraalVM, which is free to use in production and free to redistribute, at no cost, under the https://www.oracle.com/downloads/licenses/graal-free-license.html[GraalVM Free Terms and Conditions].
+
+Alternatively, you can use the https://github.com/graalvm/graalvm-ce-builds/releases/[GraalVM Community Edition]:
+
+[source, bash]
+.Java 23
+----
+sdk install java 23-graalce
+----
+
+=== GraalVM JDK Compatibility
+
+In order to support runtime compilation, one must use versions of GraalPy and the Polyglot API
+dependencies compatible with given GraalVM JDK version. If you see errors like this:
+
+[source, bash]
+----
+Your Java runtime '23.0.1+11-jvmci-b01' with compiler version '24.1.1' is incompatible with polyglot version '24.1.0'.
+----
+
+You need to override the versions of the following dependencies according to the error message:
+
+[source, text]
+----
+org.graalvm.polyglot:polyglot
+org.graalvm.python:python
+org.graalvm.python:python-embedding
+org.graalvm.python:graalpy-maven-plugin
+----
+
+== Generate a Micronaut Application Native Executable with GraalVM
+
+Starting from the GraalVM for JDK 23, Micronaut applications using GraalPy, or any Truffle language in general,
+can be compiled to native executable using GraalVM Native Image.
+
+Compiling Micronaut applications ahead of time with GraalVM significantly improves startup time and reduces
+the memory footprint of JVM-based applications.
+
+=== Native Executable metadata
```

Comment:
```
=== Required Metadata
```
I would suggest to change, we don't use the phrase like "Native Executable metadata", but "Reachability Metadata" or "Proxy Configuration"

Reply:
Could be also just "Native Image Configuration"